consensus/src/txn_manager.rs

```diff
@@ -82,48 +34,81 @@ impl TxnManager for MempoolProxy {
         let mut exclude_txns = vec![];
         for payload in exclude_payloads {
             for transaction in payload {
-                let mut txn_meta = TransactionExclusion::default();
-                txn_meta.sender = transaction.sender().into();
-                txn_meta.sequence_number = transaction.sequence_number();
-                exclude_txns.push(txn_meta);
+                exclude_txns.push(TransactionExclusion {
+                    sender: transaction.sender(),
+                    sequence_number: transaction.sequence_number(),
+                });
             }
         }
-        let mut get_block_request = GetBlockRequest::default();
-        get_block_request.max_block_size = max_size;
-        get_block_request.transactions = exclude_txns;
-        let response = self.mempool.get_block(get_block_request).await?;
-        Ok(response
-            .block
-            .unwrap_or_else(Default::default)
-            .transactions
-            .into_iter()
-            .filter_map(
-                |proto_txn| match SignedTransaction::try_from(proto_txn.clone()) {
-                    Ok(t) => Some(t),
-                    Err(e) => {
-                        security_log(SecurityEvent::InvalidTransactionConsensus)
-                            .error(&e)
-                            .data(&proto_txn)
-                            .log();
-                        None
-                    }
-                },
-            )
-            .collect())
+
+        let req = MempoolRequest::GetBlockRequest {
+            max_block_size: max_size,
+            transactions: exclude_txns,
+        };
+        let (callback_send, callback_recv) = oneshot::channel();
+
+        // send to shared mempool
+        self.mempool_channel
```

Comment:
nit: can we follow general code convention in repo and use explicit Arc::clone here instead?

Reply:
Wrapping a channel sender in Arc makes sending not possible, because we can't mutably borrow the wrapped sender because `DerefMut` trait is not implemented for the channel :/